plugins/techdocs/src/EntityPageDocs.tsx

```diff
@@ -14,18 +14,31 @@
  * limitations under the License.
  */
 
-import React from 'react';
-
-import { Entity, getCompoundEntityRef } from '@backstage/catalog-model';
+import {
+  Entity,
+  getCompoundEntityRef,
+  parseEntityRef,
+} from '@backstage/catalog-model';
 
+import React from 'react';
 import { TechDocsReaderPage } from './plugin';
-import { TechDocsReaderPageSubheader } from './reader/components/TechDocsReaderPageSubheader';
 import { TechDocsReaderPageContent } from './reader/components/TechDocsReaderPageContent';
+import { TechDocsReaderPageSubheader } from './reader/components/TechDocsReaderPageSubheader';
 
 type EntityPageDocsProps = { entity: Entity };
 
 export const EntityPageDocs = ({ entity }: EntityPageDocsProps) => {
-  const entityRef = getCompoundEntityRef(entity);
+  let entityRef = getCompoundEntityRef(entity);
+
+  if (entity.metadata.annotations?.['backstage.io/techdocs-entity']) {
```

Comment:
should probably do `const otherEntity = entity.metadata.annotations?.[TECHDOCS_EXTERNAL_ANNOTATION]` to leverage the constant and only do it once

Reply:
done